src/modules/imageresizer/ShellExtensions/dllmain.cpp

```diff
@@ -2,10 +2,163 @@
 #include "resource.h"
 #include "ShellExtensions_i.h"
 #include "dllmain.h"
+#include <interface/powertoy_module_interface.h>
+#include <common/settings_objects.h>
+#include "Settings.h"
 
 CShellExtensionsModule _AtlModule;
 
 extern "C" BOOL WINAPI DllMain(HINSTANCE hInstance, DWORD dwReason, LPVOID lpReserved)
 {
     return _AtlModule.DllMain(dwReason, lpReserved);
 }
+
+// The PowerToy name that will be shown in the settings.
+const static wchar_t* MODULE_NAME = L"ImageResizer";
+// Add a description that will we shown in the module settings page.
+const static wchar_t* MODULE_DESC = L"<no description>";
+
+class ImageResizerModule : public PowertoyModuleIface
+{
+private:
+    // The PowerToy state.
+    bool m_enabled = false;
+
+    // Load initial settings from the persisted values.
+    void init_settings();
+
+public:
+    // Constructor
+    ImageResizerModule()
+    {
+        init_settings();
+    };
+
+    // Destroy the powertoy and free memory
+    virtual void destroy() override
+    {
+        delete this;
+    }
+
+    // Return the display name of the powertoy, this will be cached by the runner
+    virtual const wchar_t* get_name() override
+    {
+        return MODULE_NAME;
+    }
+
+    // Return array of the names of all events that this powertoy listens for, with
+    // nullptr as the last element of the array. Nullptr can also be retured for empty
+    // list.
+    virtual const wchar_t** get_events() override
+    {
+        static const wchar_t* events[] = { nullptr };
+        // Available events:
+        // - ll_keyboard
+        // - win_hook_event
+        //
+        // static const wchar_t* events[] = { ll_keyboard,
+        //                                   win_hook_event,
+        //                                   nullptr };
+
+        return events;
+    }
+
+    // Return JSON with the configuration options.
+    virtual bool get_config(wchar_t* buffer, int* buffer_size) override
+    {
+        HINSTANCE hinstance = reinterpret_cast<HINSTANCE>(&__ImageBase);
+
+        // Create a Settings object.
+        PowerToysSettings::Settings settings(hinstance, get_name());
+        settings.set_description(MODULE_DESC);
+        return settings.serialize_to_buffer(buffer, buffer_size);
+    }
+
+    // Signal from the Settings editor to call a custom action.
+    // This can be used to spawn more complex editors.
+    virtual void call_custom_action(const wchar_t* action) override
+    {
+        static UINT custom_action_num_calls = 0;
+        try
+        {
+            // Parse the action values, including name.
+            PowerToysSettings::CustomActionObject action_object =
```

Comment:
Is there any point to this code?

Reply:
I think I'll change this function's implementation to {} as it still needs to be declared however it isn't being used in this case. This code was taken from the template.